bazelisk.py

```diff
@@ -68,23 +70,33 @@ def find_workspace_root(root=None):
   return find_workspace_root(new_root) if new_root != root else None
 
 
-def resolve_latest_version():
+def resolve_latest_version(offset):
   res = urlopen('https://api.github.com/repos/bazelbuild/bazel/releases').read()
-  return str(
-      max(
-          LooseVersion(release['tag_name'])
-          for release in json.loads(res.decode('utf-8'))
-          if not release['prerelease']))
-
-
-def resolve_version_label_to_number(bazelisk_directory, version):
-  if version == 'latest':
-    latest_cache = os.path.join(bazelisk_directory, 'latest_bazel')
+  all_versions = sorted((LooseVersion(release['tag_name']) for release in json.loads(res.decode('utf-8')) if not release['prerelease']), reverse=True)
+  if offset >= len(all_versions):
+    version = 'latest-{}'.format(offset) if offset else 'latest'
+    raise Exception('Cannot resolve version "{}": There are only {} Bazel '
+                    'releases.'.format(version, len(all_versions))) 
+  
+  return str(all_versions[offset])
+
+
+def resolve_version_label_to_number(version_cache_directory, version):
+  if 'latest' in version:
+    match = LATEST_PATTERN.match(version)
+    if not match:
+      raise Exception('Invalid version "{}". In addition to using a version '
+                      'number such as "0.20.0", you can use values such as '
+                      '"latest" and "latest-N", with N being a non-negative '
+                      'integer.'.format(version)) 
+    
+    offset = match.group('offset') or '0'
+    latest_cache = os.path.join(version_cache_directory, offset)
     if os.path.exists(latest_cache):
       if abs(time.time() - os.path.getmtime(latest_cache)) < ONE_HOUR:
         with open(latest_cache, 'r') as f:
           return f.read().strip()
-    latest_version = resolve_latest_version()
+    latest_version = resolve_latest_version(int(offset))
```

Comment:
I think if we slightly refactor this, we can cache the value of "latest" and then compute the "latest-N" from it, instead of making one API call per value of N and then caching all of these version numbers. WDYT?

Reply:
Yeah, we can cache the entire list of versions.